### PR TITLE
Refactor demo flow test and add basic gift shopping test case.

### PR DESCRIPTION
**Pull request by mariakleiner**
https://github.com/PolymerLabs/arcs-cdn/issues/107

### CODE REVIEW

File: dev/test/specs/starter-test.js [hunk 1]
```diff
@@ -199,30 +237,9 @@ function clickInParticles(slotName, selectors, textQuery) {
 
 describe('test basic arcs functionality', function() {
```
Comment (smalls) [inline dev/test/specs/starter-test.js]: Now this description doesn't seem right.. maybe 'test Arcs demo flows'?

Comment (smalls) [reply dev/test/specs/starter-test.js]: (low priority, though).

File: dev/test/specs/starter-test.js [hunk 2]
```diff
@@ -97,6 +97,42 @@ function loadSeleniumUtils() {
...
+  // check out some basic structure relative to the app footer
+  const footerPath = getFooterPath(); //['arc-footer', 'x-toast[app-footer]'];
+  assert.ok(pierceShadowsSingle(footerPath.slice(0, 1)).value);
+  assert.ok(pierceShadowsSingle(footerPath).value);
+}
+
+function createNewArc() {
+  // create a new arc, switch to that tab (toggling back to the first tab to
+  // reset the webdriver window state).
+  browser.waitForVisible('div[title="New Arc"]');
```
Comment (smalls) [inline dev/test/specs/starter-test.js]: Now that we have multiple tests, there's the chance that we might finish through one of the tests to browser.close() and we'd get in a weird state here.

Maybe add a precondition that there's only 1 open window at the start of this method?

File: dev/test/specs/starter-test.js [hunk 3]
```diff
@@ -231,8 +248,28 @@ describe('test basic arcs functionality', function() {
     acceptSuggestion(footerPath, 'make a reservation');
     acceptSuggestion(footerPath, 'You are free');
 
+    browser.close();
     // to drop into debug mode with a REPL; also a handy way to see the state
     // at the end of the test:
     // browser.debug();
   });
+
+  it('can use the gift shopping demo flow', function() {
```
Comment (smalls) [inline dev/test/specs/starter-test.js:259]: This method looks pretty good! It seems pretty literate. We could remove the footerPath argument (maybe?) to clean it up a little further, even.

File: dev/test/specs/starter-test.js [hunk 4]
```diff
@@ -231,8 +248,28 @@ describe('test basic arcs functionality', function() {
...
+
+    acceptSuggestion(footerPath, "Show Products from your browsing context (Minecraft Book plus 2 other items) and choose from Products recommended based on Products from your browsing context and Claire's wishlist (Book: How to Draw plus 2 other items)");
+
+    // TODO: click the "Add" buttons to move products from recommended to shortlist and
+    // (1) verify product was moved,
+    // (2) verify 'action' slot is not visible after all products were moved.
+
+    acceptSuggestion(footerPath, "Estimate arrival dates, estimate arrival dates");  // TODO: add "and buy gifts for Claire" when descriptions are fixed.
+    acceptSuggestion(footerPath, "check manufacturer information for Products from your browsing context");
+    acceptSuggestion(footerPath, "recommendations based on Products recommended based on Products from your browsing context and Claire's wishlist");
```
Comment (smalls) [inline dev/test/specs/starter-test.js]: Is there an assertion that'd make sense at the end, so we now this last suggestion worked?

**Review by smalls — APPROVED** [review]
Looks good!

File: dev/test/specs/starter-test.js (hunk 2 above)
Comment (mariakleiner) [reply dev/test/specs/starter-test.js]: done (although I'm not sure this is absolutely correct - we might want multiple arcs and hence multiple windows in the same test case later.. but it's good for now).

File: dev/test/specs/starter-test.js (hunk 4 above)
Comment (mariakleiner) [reply dev/test/specs/starter-test.js]: added some asserts, will add more later (there is a TODO), but i need additional helper methods first.

File: dev/test/specs/starter-test.js (hunk 3 above)
Comment (mariakleiner) [reply dev/test/specs/starter-test.js:259]: done!

File: dev/test/specs/starter-test.js (hunk 1 above)
Comment (mariakleiner) [reply dev/test/specs/starter-test.js]: done

**Pull request merged**